Library/Homebrew/extend/ENV.rb

```diff
@@ -4,11 +4,7 @@
 require "extend/ENV/super"
 
 def superenv?
-  if OS.linux?
-    ARGV.env == "super" || ARGV.homebrew_developer? && ARGV.env != "std"
```

Comment:
AND has higher precedence than OR. http://phrogz.net/programmingruby/language.html#table_18.4